src/popup/modules/InitQrCode.js

```diff
@@ -85,3 +85,38 @@ export const initiationProcess = Promise.all([qrCreatorInit, userInterfaceInit])
 }).catch((error) => {
     Logger.logError(error);
 });
+
+/**
+ * Queries Tabs API for 20 times with a delay of 300ms until the tabs have a defined url.
+ * Will reject if url is not defined.
+ * @returns {Promise}
+ */
+async function getsTabWithValidUrl() {
+    let retryCount = 0;
+    const maxRetries = 20;
+    const delay = 300;
+
+    try {
+        return await retryPromise(async () => {
+            if (retryCount === maxRetries) {
+                throw new Error("No tabs with url.");
+            }
+
+            try {
+                retryCount++;
+
+                const tabs = await queryBrowserTabs();
+
+                if (tabs && tabs[0].url) {
+                    return tabs;
```

Comment:
Hmm, the function says `getsTabWithValidUrl`, so it says it returns one tab, but here it returns all of them.

So I guess you need some small re-factoring to adjust that behaviour. I'd prefer if it really just returns one tab. Of course, you may adjust/delete some functions of the `QrCreator` then.